src/init.cpp

```diff
@@ -1843,7 +1851,6 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
         if (!Lookup(i2psam_arg, addr, 7656, fNameLookup) || !addr.IsValid()) {
             return InitError(strprintf(_("Invalid -i2psam address or hostname: '%s'"), i2psam_arg));
         }
-        SetReachable(NET_I2P, true);
```

Comment:
What's the reason for deleting this line?

Reply:
Same logic as with Tor - avoid explicitly calling `SetReachable(..., true)` because:

1. Everything is reachable by default (all elements in `vfLimited[]` are `false` by default), so if the network is reachable anyway, `SetReachable(..., true)` is a noop.
2. If the network is not reachable by the time this code executes, that means that `-onlynet=` was specified and the network in question is not among the lucky ones being listed. In this case, respect the `-onlynet=` setting.

Reply:
Why even bother doing all these checks if the network was limited in the first place? It just makes it a bit harder to understand, especially in the context of this diff, but no big deal.

Reply:
Ah, here's the answer: `This does not apply to hosts given to `-connect`, `-addnode`,
`addnode` RPC, dns seeds, `-seednodes`.`

Reply:
Yeah, because a proxy was provided, it is possible to manually open connections to those networks.